app/src/main/java/pl/futurecollars/invoicing/db/InvoiceRepository.java

```diff
@@ -0,0 +1,20 @@
+package pl.futurecollars.invoicing.db;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import pl.futurecollars.invoicing.model.Invoice;
+
+public interface InvoiceRepository {
+
+    Invoice save(Invoice invoice);
+
+    Optional<Invoice> getById(UUID id) throws IOException;
+
+    List<Invoice> getAll();
+
+    Invoice update(UUID id, Invoice updatedInvoice) throws IOException;
```

Comment:
interfejsy nie powinny mieć nigdy sygnatur z 'throws [..]'.
To wymusza na każdej implementacji dostosowanie się i 'zapaskudzenie' sygnatury metody wlasnie tym 'throws ...', a przecież nie każda implementacja musi ten throws rzucać.

Stąd też o wiele lepszym rozwiązaniem jest przechwycenie checked exceptionow **tak szybko jak to tylko możliwe** i zamiana ich na unchecked exception - do poczytania: https://phauer.com/2015/checked-exceptions-are-evil/